src/components/searchView.jsx

```diff
@@ -1,15 +1,99 @@
 import React from "react";
+import RouteView from "./routeView";
+import LocationMap from "./locationMap";
 
 class SearchView extends React.Component {
   constructor (props) {
     super(props);
-    this.state = {};
+    this.state = {
+      routes: [],
+      place: undefined,
+      error: false,
+      loading: true,
+      visible: false,
+      featured: 0,
+      routeView: false
+    };
+  }
+
+  componentDidMount () {
+    fetch("http://localhost:3060/api/v1/routes")
+      .then(res => res.json())
+      .then(json => {
+        this.setState({
+          routes: json.body,
+          loading: false
+        });
+      })
+      .catch(() => {
+        this.setState({ error: true });
+      });
+  }
+
+  showPlace (key, routeId) {
+    fetch(`http://localhost:3060/api/v1/places/${routeId}`)
+      .then(res => res.json())
+      .then(json => {
+        this.setState({
+          place: json.body,
+          featured: key,
+          visible: true });
+      })
+      .catch(() => {
+        this.setState({ error: true });
+      });
+  }
+
+  goRoute (e) {
+    e.preventDefault();
+    this.setState({ routeView: true });
   }
 
   render () {
+    let routes = this.state.routes;
+    let routesArr = [];
+    let showed;
+    let featured;
+
+    for (let key in routes) {
+      routesArr.push(
+        <div className="route" key={key} onClick={() => this.showPlace(key, routes[key].id)}>
+          <p>{routes[key].name}</p>
+          <p>{routes[key].description}</p>
+        </div>);
+    }
+
+    if (this.state.loading) {
+      showed = <div></div>;
+    } else {
+      showed = routesArr;
```

Comment:
I'm not sure I like the name "showed" - can you find something a little more descriptive? Even "routes" would be better.

Reply:
sure!